### PR TITLE
Add Append method to ResponseWriter

### PR DESCRIPTION
**Pull request by mzhou**
This mirrors the Response::append_body method in the Rust SDK: https://docs.rs/fastly/0.9.5/fastly/http/struct.Response.html#method.append_body

### CODE REVIEW

File: fsthttp/response.go [hunk 1]
```diff
@@ -111,6 +111,10 @@ type ResponseWriter interface {
 	//
 	// To have an effect on the response, this must be called before any call to Write() or WriteHeader().
 	SetManualFramingMode(bool)
+
+	// Append a body onto the end of this response.
+	// This operation is performed in amortized constant time, and so should always be preferred to reading an entire body and then writing the same contents to another body.
```
Comment (joeshaw) [inline fsthttp/response.go:116]: I think we need to make clear that the reader has to come from a Response's Body field, or make the function work with non-`HTTPBody` types [a la `Request.Send()`](https://github.com/fastly/compute-sdk-go/blob/main/fsthttp/request.go#L317-L333).

I might also make it more explicit that this is preferable to `io.Copy`, which I think most Go programmers will understand.  Something like, "This operation is performed in amortized constant time, and so should always be preferred to directly copying a body with io.Copy".

File: fsttest/recorder.go [hunk 2]
```diff
@@ -48,3 +49,9 @@ func (r *ResponseRecorder) Close() error {
 // SetManualFramingMode is a no-op on ResponseRecorder.  It exists to
 // satisfy the fsthttp.ResponseWriter interface.
 func (r *ResponseRecorder) SetManualFramingMode(v bool) {}
+
+// Append is a no-op on ResponseRecorder.  It exists to
+// satisfy the fsthttp.ResponseWriter interface.
+func (r *ResponseRecorder) Append(other io.ReadCloser) error {
+	return nil
```
Comment (joeshaw) [inline fsttest/recorder.go:56]: This shouldn't be a no-op.  We'll need to copy the bytes into `r.Body` here.

Comment (joeshaw) [reply fsttest/recorder.go:56]: Lots of the existing integration tests use `io.Copy` -- it'd be good to update at least one of them to use `Append` instead to exercise this.

**Comment by mzhou** [comment]
Thanks for the feedback. Updated.

**Review by joeshaw — APPROVED** [review]
👍

**Comment by dgryski** [comment]
Thanks.

**Pull request merged**